tests/testthat/test_merge.R

```diff
@@ -1,32 +1,45 @@
 
 # load test data (df1, df2, pkList, fkList)
-# load(file = "tests/testthat/helper_data.rda")
-load(file = "helper_data.rda")
+load(file = test_path("helper_data.rda"))
 
+mod_dat <- data.frame(ID1 = c(1, 3, 4), V2 = c(8, 10, 9))
+df3 <- suppressMessages(updateMeta(df1, mod_dat))
 
-test_that("Merge two GADSdat objects",{
-  df3 <- df1
-  mod_dat <- df3$dat
-  mod_dat[, "v3"] <- c(8, 7)
-  df3 <- suppressMessages(updateMeta(df3, mod_dat))
+
+test_that("Errors for merge", {
   expect_error(merge(df1, df3, by = 2))
   expect_error(merge(df1, df3, by = "x2"))
   expect_error(merge(df1, df1, by = "V1"))
-  out <- merge(df1, df3, by = "V1")
-  expect_equal(out$dat[, 3], c(8, 7))
+})
+
+test_that("Merge two GADSdat objects",{
+  out <- merge(df1, df3, by = "ID1")
+  expect_equal(out$dat[, "V1"], c(3, 5, NA, NA))
+  expect_equal(out$dat[, "V2"], c(8, NA, 10, 9))
   expect_equal(nrow(out$labels), 3)
 })
 
-test_that("Inner join two GADSdat objects", {
-  df3 <- df1
-  mod_dat <- df3$dat
-  mod_dat[, "v3"] <- c(8, 7)
-  df3 <- suppressMessages(updateMeta(df3, mod_dat))
-  df3$dat[3, ] <- c(1, 2, 5)
+test_that("Merge two GADSdat objects while setting default missing value",{
+  out <- merge(df1, df3, by = "ID1", missingValue = -99, missingValLabel = "missing")
+  expect_equal(out$dat[, "V1"], c(3, 5, -99, -99))
+  expect_equal(out$dat[, "V2"], c(8, -99, 10, 9))
+  expect_equal(nrow(out$labels), 3)
 
-  out <- merge(df1, df3, by = "V1", all = F)
-  expect_equal(dim(out$dat), c(2, 3))
+  # with existing NAs
+  df1b <- df1
+  df1b$dat[1, "V1"] <- NA
+  df3b <- df3
+  df3b$dat[2:3, "V2"] <- NA
+  out2 <- merge(df1b, df3b, by = "ID1", missingValue = -99, missingValLabel = "missing")
+  expect_equal(out2$dat[, "V1"], c(NA, 5, -99, -99))
+  expect_equal(out2$dat[, "V2"], c(8, -99, NA, NA))
+  expect_equal(nrow(out2$labels), 3)
+  expect_equal(out2$labels[2:3, "value"], c(-99, -99))
+  expect_equal(out2$labels[2:3, "valLabel"], c("missing", "missing"))
+  expect_equal(out2$labels[2:3, "missings"], c("miss", "miss"))

```

Comment:
Nit: Maybe, one test battery could be added where `missingValLabel `is `NULL` (I first thought, the `missingValLabel` was required when `missingValue` was set):

```r
  # with existing NAs without valLabel
  df1c <- df1
  df1c$dat[1, "V1"] <- NA
  df3c <- df3
  df3c$dat[2:3, "V2"] <- NA
  out3 <- merge(df1c, df3c, by = "ID1", missingValue = -99)
  expect_equal(out3$dat[, "V1"], c(NA, 5, -99, -99))
  expect_equal(out3$dat[, "V2"], c(8, -99, NA, NA))
  expect_equal(nrow(out3$labels), 3)
  expect_equal(out3$labels[2:3, "value"], c(-99, -99))
  expect_equal(out3$labels[2:3, "valLabel"], c(NA_character_, NA_character_))
  expect_equal(out3$labels[2:3, "missings"], c("miss", "miss"))
```

Reply:
Agree and modified as suggested (but with some simplified code, re-using the data of the previous test case)!